src/sidebar/components/excerpt.js

```diff
@@ -1,178 +1,189 @@
-        }
-        // Even though this change happens outside the framework, we use
-        // $digest() rather than $apply() here to avoid a large number of full
-        // digest cycles if many excerpts update their overflow state at the
-        // same time. The onCollapsibleChanged() handler, if any, is
-        // responsible for triggering any necessary digests in parent scopes.
-        $scope.$digest();
-      },
-    },
-    window.requestAnimationFrame
+/**
+ * A container which truncates its content when they exceed a specified height.
+ *
+ * The collapsed state of the container can be handled either via internal
+ * controls (if `inlineControls` is `true`) or by the caller using the
+ * `collapse` prop.
+ */
+function Excerpt({
+  children,
+  collapse = false,
+  collapsedHeight,
+  inlineControls = true,
+  onCollapsibleChanged = noop,
+  onToggleCollapsed = noop,
+  overflowHysteresis = 0,
+  settings = {},
+}) {
+  const [collapsedByInlineControls, setCollapsedByInlineControls] = useState(
+    true
   );
 
-  this.contentStyle = overflowMonitor.contentStyle;
+  // Container for the excerpt's content.
+  const contentElement = useRef(null);
+
+  // Measured height of `contentElement` in pixels.
+  const [contentHeight, setContentHeight] = useState(0);
+
+  // Update the measured height of the content after the initial render and
+  // when the size of the content element changes.
+  const updateContentHeight = useCallback(() => {
+    const newContentHeight = contentElement.current.clientHeight;
+    setContentHeight(newContentHeight);
+
+    const isCollapsible =
+      newContentHeight > collapsedHeight + overflowHysteresis;
+    onCollapsibleChanged({ collapsible: isCollapsible });
+  }, [collapsedHeight, onCollapsibleChanged, overflowHysteresis]);
+
+  useLayoutEffect(() => {
+    const cleanup = observeElementSize(
+      contentElement.current,
+      updateContentHeight
+    );
+    updateContentHeight();
+    return cleanup;
+  }, [updateContentHeight]);
+
+  // Render the (possibly truncated) content and controls for
+  // expanding/collapsing the content.
+  const overflowing = contentHeight > collapsedHeight + overflowHysteresis;
+  const isCollapsed = inlineControls ? collapsedByInlineControls : collapse;
+  const isExpandable = overflowing && isCollapsed;
+
+  const contentStyle = {};
+  if (contentHeight !== 0) {
+    contentStyle['max-height'] = isExpandable ? collapsedHeight : contentHeight;
+  }
 
-  // Listen for document events which might affect whether the excerpt
-  // is overflowing, even if its content has not changed.
-  $element[0].addEventListener(
-    'load',
-    overflowMonitor.check,
-    false /* capture */
+  const setCollapsed = collapsed =>
+    inlineControls
+      ? setCollapsedByInlineControls(collapsed)
+      : onToggleCollapsed(collapsed);
+
+  return (
+    <div className="excerpt" style={contentStyle}>
+      <div test-name="excerpt-content" ref={contentElement}>
```

Comment:
It has no special meaning. I just added it as a convenience to be able to find this element in tests, since the `<div>` has no other distinguishing attributes.